src/chatlist.rs

```diff
@@ -202,9 +202,9 @@ impl Chatlist {
                                   AND (hidden=0 OR state=?))
                  WHERE c.id>9
                    AND c.blocked=0
-                   AND c.archived=0
+                   AND NOT c.archived=1
```

Comment:
I'm not saying this should change in this PR.  But that we need this kind of code makes me a little sad.  It breaks the abstraction ToSql/FromSql give by now needing to know how they are implemented.  But perhaps only going full ORM would solve something like this so not worth breaking our head over now.

Reply:
i think we can also use ChatVisibility::Archived here, thanks for the hint.

Reply:
changed it to used ChatVisibility::Archived and ChatVisibility::Pinned here, i think it is an overall improvement in this case.
however, as the usage and the definition of the value are visually far away now, i do not think this is a good pattern for all cases, also as if we do not have many problems with sqlite statements at all.